### ISSUE TITLE
Improvement: Maintain wire connection to code blocks when renaming variables or if syntax exception is thrown

### ISSUE THREAD

**Issue by ThomasMahon** [DynamoDS/Dynamo#7384]
## Dynamo version

1.2.1

## Operating system

Win 10

## Improvement Suggestion
I'm finding a behaviour in Dynamo a nuisance: whenever a variable name is changed in a code block which has a wire connected, the wire will be automatically disconnect. I simply want to rename the variable (to rename the port). I don't need the wire disconnected automatically. Can this behaviour be changed? 

**Before:**
![image](https://cloud.githubusercontent.com/assets/18601851/20790346/e8006d3c-b7af-11e6-935c-1687ca073b43.png)

**After:** changing variable name, wire is automatically disconnected
![image](https://cloud.githubusercontent.com/assets/18601851/20790362/f3ac326a-b7af-11e6-98c6-6d9bf8c8370a.png)

Also, another issue with CB is when an exception is thrown due to a syntax error. In these cases ALL wires get disconnected. This needs to be improved:

**Before:**
![image](https://cloud.githubusercontent.com/assets/18601851/20790446/3ff66dde-b7b0-11e6-8ee4-06f684411eab.png)

**After:** a syntax error causes all wires to disconnect
![image](https://cloud.githubusercontent.com/assets/18601851/20790472/5833d292-b7b0-11e6-9896-06ac1f7788ea.png)

**Comment by kronz** [comment]
@ke-yu I believe this is being addressed?

**Comment by aparajit-pratap** [comment]
https://github.com/DynamoDS/Dynamo/pull/7401 addresses maintaining ports and connections on error.

**Comment by Curty102** [comment]
In #6510 near the end I comment about the option to rename variables either specifically, thus creating a new variable (current default behavior), or GLOBALLY, perhaps via right-click, which would change the name of that variable everywhere it is used.  There would be some work to govern how far the rename propagates.

**Comment by aparajit-pratap** [comment]
We have addressed the issue of maintaining connections for code block errors. There are no plans as of yet to maintain them for renamed variables. Please create the latter as a new "Improvement request" if necessary. Closing this issue for now. @ThomasMahon @kronz

**Comment by jhubers** [comment]
I support this request. It starts anoying me too.
